### DISCUSSION TITLE
How to add npm packages to directus docker image?

### DISCUSSION THREAD

**Discussion by navanshu** [directus/directus#17483] [Archive]
I am trying to install some sdks inside directus official image

```Docker
FROM directus/directus:latest

USER node
COPY ./[]/dist/index.js /directus/extensions/endpoints/[]/index.js
COPY ./[]/dist/index.js /directus/extensions/hooks/[]/index.js

WORKDIR /directus

RUN npm install nanoid couchbase
```

Then I use this container to generate an Image and then use it, but when I run docker compose -up, I get this error
```console
directus  | /usr/local/bin/docker-entrypoint.sh: exec: line 11: cd: not found
```

I can't figure out, what I am doing wrong, plus how is standard image of directus is made?
If I understand that, I can directly update the source code.

**Comment by claytongulick** [comment]
I know that the official recommendation is to run Directus in the docker container, but I've found it to be much less complex and easy to debug to just run it via npx directly on a VM.

My repo has directus as a dependency in package.json, and to start directus I just do ```npx directus start```. My project structure has /extensions folder at the top level, and all those deps are in the top level package.json.

I also do a monorepo and use pnpm workspaces, so my entire deployment process is just ```git pull```, ```pnpm install```. Keeps it simple and easy to debug issues.

**Reply by navanshu** [reply]
Yeah, I thought of that, make a few scripts to take care of updates and migrations.

**Reply by abdonrd** [reply]
This discussion may help you:
- https://github.com/directus/directus/discussions/17310

**Comment by marchiolidev** [comment]
For those facing the same problem, I found a way to do it with a custom Dockerfile based on the official one:

```
FROM directus/directus
WORKDIR /directus

#Directus uses pnpm as a package manager, but the official docker image doesn't come with it installed and the node user doesn't have permission to install global packages, so we switch to root and install it
USER root
RUN npm install -g pnpm@9.0.6 

#We switch back to the node user and use pnpm to install our package
USER node
RUN pnpm i mysql2

#In this example, we're changing the default driver for mysql to mysql2, so we need to run the following line to make this change effective. You probably won't need it but here it is just in case.
RUN sed -i -e 's/Client_MySQL/Client_MySQL2/' node_modules/@directus/api/dist/database/index.js
```
